seed/models/data_quality.py

```diff
@@ -702,76 +735,63 @@ def _check(self, rules, row):
                 if (rule.table_name, rule.field) not in self.column_lookup:
                     # If the rule is not in the column lookup, then it may have been a required
                     # field that wasn't mapped
-                    if rule.required:
+                    if rule.condition == Rule.RULE_REQUIRED:
                         self.add_result_missing_req(row.id, rule, display_name, value)
                         label_applied = self.update_status_label(label, rule, linked_id, row.id)
                 elif value is None or value == '':
-                    # Empty fields
-                    if rule.required:
-                        self.add_result_missing_and_none(row.id, rule, display_name, value)
-                        label_applied = self.update_status_label(label, rule, linked_id, row.id)
-                    elif rule.not_null:
-                        self.add_result_is_null(row.id, rule, display_name, value)
+                    if rule.severity == Rule.SEVERITY_ERROR or rule.severity == Rule.SEVERITY_WARNING:
```

Comment:
What's the motivation behind this initial severity level check?

Couldn't there be a case where users would consider it "Valid Data" and just want a label automatically applied if a field is null or missing?

Reply:
We had a conversation earlier on about data quality check should only report errors and warning. It does not report valid data. However, a label is mandatory for a rule set for filtering valid data on the UI:
<img width="1036" alt="Screen Shot 2020-05-07 at 1 44 25 PM" src="https://user-images.githubusercontent.com/49968203/81337837-e4eda700-9068-11ea-94e4-0ab01f456699.png">

Reply:
You mean users see a null value as a valid value?